### DISCUSSION TITLE
Sales orders split into two manufacturing orders with the option not to allow splitting.

### DISCUSSION THREAD

**Discussion by salvastp** [frePPLe/frepple#622] [Q&A]
Hello everyone,

I have a sales order for an item with an operation defined exclusively for it. It has no other dependent operations, and the system generates two consecutive work orders in time.

This shouldn't happen if the option to allow splitting sales orders into multiple orders is set to "false," and it also doesn't make much sense to split the orders and place them one after the other.

![imagen](https://github.com/frePPLe/frepple/assets/129965662/a3ea2416-44aa-44de-a86a-dd81c902b7b7)
![manufacturing orders](https://github.com/frePPLe/frepple/assets/129965662/773af4e5-915d-4364-8f78-ec0487d6e23b)
![parameters](https://github.com/frePPLe/frepple/assets/129965662/93b5b7aa-2d5c-48cd-98f4-8c83f6878472)

**Comment by jdetaeye** [comment]
There are some legitimate cases where a split is possible by the algorithm, even with this parameter set to false. You would need to run with the parameter "plan.loglevel" set to 9 to understand why it's splitting the requirement in multiple parts.

**Reply by salvastp** [reply]
I'm sorry, but I don't quite understand the log. Is there any way to prevent the production order from being split?
Thanks in advance


Planning demand '347071-10': 670100010, 2023-11-22T00:10:00, 308
  Demand '347071-10' asks: 308  2021-02-26T00:10:00
    Operation '347071-10' is asked: 308  2021-02-26T00:10:00
    Operation '347071-10' answers: 0  2023-11-22T08:16:00
  Demand '347071-10' gets answer: 0  2023-11-22T08:16:00  0  0
  Demand '347071-10' asks: 308  2023-11-22T08:16:00
    Operation '347071-10' is asked: 308  2023-11-22T08:16:00
    Operation '347071-10' evaluates alternate '236': cost 0, penalty 0
      Operation '347071-10' chooses alternate '236' MINPENALTY
      Resource '236' is asked: 1  2023-11-22T08:00:00 / 2023-11-22T08:16:00
      Resource '236' answers: 1  2023-11-22T08:16:00
    Operation '347071-10' answers: 30
  Demand '347071-10' gets answer: 30  30  0
  Demand '347071-10' plans coordination.
  Demand '347071-10' asks: 278  2023-11-22T08:16:00
    Operation '347071-10' is asked: 278  2023-11-22T08:16:00
    Operation '347071-10' evaluates alternate '236': not available before 2023-11-22T08:32:00
      Alternate load doesn't find supply on any alternate: not available before 2023-11-22T08:32:00
    Operation '347071-10' answers: 0  2023-11-22T08:32:00
  Demand '347071-10' gets answer: 0  2023-11-22T08:32:00  60  0
  Demand '347071-10' asks: 278  2023-11-22T08:32:00
    Operation '347071-10' is asked: 278  2023-11-22T08:32:00
    Operation '347071-10' evaluates alternate '236': not available before 2023-11-22T08:48:00
      Alternate load doesn't find supply on any alternate: not available before 2023-11-22T08:48:00
    Operation '347071-10' answers: 0  2023-11-22T08:48:00
  Demand '347071-10' gets answer: 0  2023-11-22T08:48:00  60  0
  Demand '347071-10' asks: 278  2023-11-22T08:48:00
    Operation '347071-10' is asked: 278  2023-11-22T08:48:00
    Operation '347071-10' evaluates alternate '236': not available before 2023-11-22T09:04:00
      Alternate load doesn't find supply on any alternate: not available before 2023-11-22T09:04:00
    Operation '347071-10' answers: 0  2023-11-22T09:04:00
  Demand '347071-10' gets answer: 0  2023-11-22T09:04:00  60  0
  Demand '347071-10' asks: 278  2023-11-22T09:04:00
    Operation '347071-10' is asked: 278  2023-11-22T09:04:00
    Operation '347071-10' evaluates alternate '236': not available before 2023-11-22T09:20:00
      Alternate load doesn't find supply on any alternate: not available before 2023-11-22T09:20:00
    Operation '347071-10' answers: 0  2023-11-22T09:20:00
  Demand '347071-10' gets answer: 0  2023-11-22T09:20:00  60  0
  Demand '347071-10' asks: 278  2023-11-22T09:20:00
    Operation '347071-10' is asked: 278  2023-11-22T09:20:00
    Operation '347071-10' evaluates alternate '236': not available before 2023-11-22T09:36:00
      Alternate load doesn't find supply on any alternate: not available before 2023-11-22T09:36:00
    Operation '347071-10' answers: 0  2023-11-22T09:36:00
  Demand '347071-10' gets answer: 0  2023-11-22T09:36:00  60  0
  Demand '347071-10' asks: 278  2023-11-22T09:36:00
    Operation '347071-10' is asked: 278  2023-11-22T09:36:00
    Operation '347071-10' evaluates alternate '236': not available before 2023-11-22T09:52:00
      Alternate load doesn't find supply on any alternate: not available before 2023-11-22T09:52:00
    Operation '347071-10' answers: 0  2023-11-22T09:52:00
  Demand '347071-10' gets answer: 0  2023-11-22T09:52:00  60  0
  Demand '347071-10' asks: 278  2023-11-22T09:52:00
    Operation '347071-10' is asked: 278  2023-11-22T09:52:00
    Operation '347071-10' evaluates alternate '236': not available before 2023-11-22T10:08:00
      Alternate load doesn't find supply on any alternate: not available before 2023-11-22T10:08:00
    Operation '347071-10' answers: 0  2023-11-22T10:08:00
  Demand '347071-10' gets answer: 0  2023-11-22T10:08:00  60  0
  Demand '347071-10' asks: 278  2023-11-22T10:08:00
    Operation '347071-10' is asked: 278  2023-11-22T10:08:00
    Operation '347071-10' evaluates alternate '236': not available before 2023-11-22T10:24:00
      Alternate load doesn't find supply on any alternate: not available before 2023-11-22T10:24:00
    Operation '347071-10' answers: 0  2023-11-22T10:24:00
  Demand '347071-10' gets answer: 0  2023-11-22T10:24:00  60  0
  Demand '347071-10' asks: 278  2023-11-22T10:24:00
    Operation '347071-10' is asked: 278  2023-11-22T10:24:00
    Operation '347071-10' evaluates alternate '236': not available before 2023-11-22T10:40:00
      Alternate load doesn't find supply on any alternate: not available before 2023-11-22T10:40:00
    Operation '347071-10' answers: 0  2023-11-22T10:40:00
  Demand '347071-10' gets answer: 0  2023-11-22T10:40:00  60  0
  Demand '347071-10' asks: 278  2023-11-22T10:40:00
    Operation '347071-10' is asked: 278  2023-11-22T10:40:00
    Operation '347071-10' evaluates alternate '236': not available before 2023-11-22T10:44:16
      Alternate load doesn't find supply on any alternate: not available before 2023-11-22T10:44:16
    Operation '347071-10' answers: 0  2023-11-22T10:44:16
  Demand '347071-10' gets answer: 0  2023-11-22T10:44:16  60  0
  Demand '347071-10' asks: 278  2023-11-22T10:44:16
    Operation '347071-10' is asked: 278  2023-11-22T10:44:16
    Operation '347071-10' evaluates alternate '236': cost 0, penalty 0
      Operation '347071-10' chooses alternate '236' MINPENALTY
      Resource '236' is asked: 1  2023-11-22T08:16:00 / 2023-11-22T10:44:16
      Resource '236' answers: 1  2023-11-22T10:44:16
    Operation '347071-10' answers: 278
  Demand '347071-10' gets answer: 278  338  0

**Comment by jdetaeye** [comment]
Not that easy to interpret the frepple log files.  
 
>Demand '347071-10' asks: 308 2023-11-22T08:16:00
>Operation '347071-10' is asked: 308 2023-11-22T08:16:00
>Operation '347071-10' evaluates alternate '236': cost 0, penalty 0
>Operation '347071-10' chooses alternate '236' MINPENALTY
>Resource '236' is asked: 1 2023-11-22T08:00:00 / 2023-11-22T08:16:00
>Resource '236' answers: 1 2023-11-22T08:16:00
>Operation '347071-10' answers: 30
>Demand '347071-10' gets answer: 30 30 0

This section indicates that it's related to capacity. Only 30 units fit within the available time slot of capacity.

If the operation wouldn't have been an alternate with MINPENALTY criterium, I think the demand would have stayed in 1 MO.  Try leaving the priority and search_mode fields of operation 236 blank. 
The case where a requirement is distributed across multiple alternates is one of these legitimate cases where we accept splitting.

**Reply by salvastp** [reply]
Good morning,

After reviewing the data, I can't find anything indicating a problem:

    The operation is the first one scheduled on resource "236."
    The date is today at 8:00, and it has 8 hours available on the resource.
    The second operation is scheduled exactly after the first one (as seen in the image).
    There is no alternative type configured.
    The next operation belongs to the same batch and is configured exactly the same, and it is scheduled correctly afterward.
    It might be a good idea to have an option in operations to force the algorithm to "not" split the operation.

Thank you for your assistance.
![FREPPLE](https://github.com/frePPLe/frepple/assets/129965662/41b90138-8198-407a-84ec-ff340a147268)
